tsuru/resource_tsuru_app.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2021 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tsuru
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	tsuru_client "github.com/tsuru/go-tsuruclient/pkg/tsuru"
+)
+
+func resourceTsuruApplication() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Tsuru Application",
+		CreateContext: resourceTsuruApplicationCreate,
+		UpdateContext: resourceTsuruApplicationUpdate,
+		ReadContext:   resourceTsuruApplicationRead,
```

Comment:
@pedrokiefer could we reuse the same read function of datasource ?

Reply:
I couldn't find any provider that would do that, I don't think that's a good idea.